java/c/src/main/java/org/apache/arrow/c/ArrayImporter.java

```diff
@@ -141,9 +141,10 @@ private List<ArrowBuf> importBuffers(ArrowArray.Snapshot snapshot) {
     for (long bufferPtr : buffers) {
       ArrowBuf buffer = null;
       if (bufferPtr != NULL) {
-        // TODO(roee88): an API for getting the size for each buffer is not yet
-        // available
-        buffer = new ArrowBuf(referenceManager, null, Integer.MAX_VALUE, bufferPtr);
+        // See ARROW-17657: [C++] C data interface: Add field for byte length of buffers in ArrowArray
```

Comment:
I'm not sure I understand this comment, is it a TODO?

Reply:
```suggestion
        // TODO(ARROW-17657): Add field for byte length of buffers in ArrowArray
```

Reply:
Now that I read ARROW-17657, it doesn't seem desirable at all. I would recommend closing the JIRA and removing this comment :-)

Reply:
How about opening a ticket with title like "Add Java API to get imported buffer size"? Since ARROW-17657 indeed introduces breaking change.